Train.py

```diff
@@ -31,9 +31,23 @@ def main():
     torch.cuda.set_device(ARGS.gpu)
     torch.cuda.device(ARGS.gpu)
 
-    net = SqueezeNetShallow().cuda()
+    net = Z2Bala().cuda()
     criterion = torch.nn.MSELoss().cuda()
-    optimizer = torch.optim.Adagrad(net.parameters(), lr=.1)
+    
+    base_lr = .005 # z2caffe = .005
```

Comment:
Please read the PyTorch docs on optimizers, they already do all of what was implemented here in a much more efficient manner (in C code rather than Python). Find these docs here:

http://pytorch.org/docs/master/optim.html